src/main/java/org/fungover/haze/HazeList.java

```diff
@@ -270,4 +271,39 @@ private static String getKey(List<String> inputList) {
             key = inputList.get(1);
         return key;
     }
+
+        public String lSet(List<String> inputlist){
+        if (inputlist.size() < 4){
+            return "-Err Wrong number of arguments for LSET\r\n";
+        }
+        String key = getKey(inputlist);
+
+        if (!hazeDatabase.containsKey(key)){
+            return "-Err Key does not exist\r\n";
+        }
+            int index;
+            if (inputlist.get(2).equals("-1")) {
+
+                List<String> list = getValueAsList(hazeDatabase.getValue(key));
+                index = list.size() - 1;
+            } else {
+                try {
+                    index = Integer.parseInt(inputlist.get(2));
+                } catch (NumberFormatException e) {
+                    return "-Err invalid index\r\n";
```

Comment:
Cover test for this and also for 277 and 282.

Reply:
Will look in to that, thanks for good feedback